platform/default/online_file_source.cpp

```diff
@@ -176,6 +167,66 @@ class OnlineFileSource::Impl {
         }
     }
 
+    struct PendingRequests {
+        // initialize two pending queues: a high priority queue for "regular" requests and a low priority queue for offline download requests
+        // the high priority requests are in prioritizedPendingRequests*[0]
+        // the low priority requests are in prioritizedPendingRequests*[1]
+        // context: https://github.com/mapbox/mapbox-gl-native/issues/12655
+        PendingRequests() : prioritizedPendingRequestsList(2), prioritizedPendingRequestsMap(2) {}
+
+        std::vector<std::list<OnlineFileRequest*>> prioritizedPendingRequestsList;
+        std::vector<std::unordered_map<OnlineFileRequest*, std::list<OnlineFileRequest*>::iterator>> prioritizedPendingRequestsMap;
```

Comment:
The original code used a list and a map with request pointer -> list iterator and was added in https://github.com/mapbox/mapbox-gl-native/commit/f5b521ffb51e396be322350a204b835a0bf0af08, but I'm not convinced that we need two separate data structures here. Given that the number of requests we have in the queue is usually pretty small (<100), a linear search in the queue might be just as fast and would simplify the logic quite a bit.